planner/core/rule_join_elimination.go

```diff
@@ -173,25 +183,28 @@ func (o *outerJoinEliminator) isDuplicateAgnosticAgg(p LogicalPlan) (_ bool, col
 	return true, cols
 }
 
-func (o *outerJoinEliminator) doOptimize(p LogicalPlan, aggCols []*expression.Column, parentSchema *expression.Schema) LogicalPlan {
+func (o *outerJoinEliminator) doOptimize(p LogicalPlan, aggCols []*expression.Column, parentSchema *expression.Schema) (LogicalPlan, error) {
 	// check the duplicate agnostic aggregate functions
 	if ok, newCols := o.isDuplicateAgnosticAgg(p); ok {
 		aggCols = newCols
 	}
 
 	newChildren := make([]LogicalPlan, 0, len(p.Children()))
 	for _, child := range p.Children() {
-		newChild := o.doOptimize(child, aggCols, p.Schema())
+		newChild, err := o.doOptimize(child, aggCols, p.Schema())
+		if err != nil {
+			return nil, err
+		}
 		newChildren = append(newChildren, newChild)
 	}
 	p.SetChildren(newChildren...)
 	join, isJoin := p.(*LogicalJoin)
 	if !isJoin {
-		return p
+		return p, nil
 	}
 	return o.tryToEliminateOuterJoin(join, aggCols, parentSchema)
 }
 
 func (o *outerJoinEliminator) optimize(p LogicalPlan) (LogicalPlan, error) {
-	return o.doOptimize(p, nil, nil), nil
+	return o.doOptimize(p, nil, nil)
```

Comment:
It looks like we don't need `doOptimize` function. Could you pls remove it?

Reply:
Sure, I will remove it.

Reply:
It seems that we need the second and the thrid parameters.  Reomve it maybe not an good idea.

Reply:
Yeah, you are right.